### PR TITLE
[DO NOT MERGE] Support for one shot polling mode 

### PR DESCRIPTION
**Pull request by reshtitus**
* Added support for one shot mode in lps33hw and lps33thw drivers.

### CODE REVIEW

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 1]
```diff
@@ -1025,59 +1080,124 @@ static int
...
+	      std.std_temp_is_valid = 1;
+	      rc = data_func(sensor, data_arg, &std,
+			     SENSOR_TYPE_TEMPERATURE);
+	  }
+	  if (!(type & SENSOR_TYPE_TEMPERATURE) && !(type & SENSOR_TYPE_PRESSURE)) {
+	      return SYS_EINVAL;
+	  }
+      }
+  }
+  else{
```
Comment (vrahane) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: Indentation for else, please check the CODING_STYLE doc

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 2]
```diff
@@ -1025,59 +1080,124 @@ static int
...
+	  /* Stream read */
+	  lps33hw->pdd.user_handler = data_func;
+	  lps33hw->pdd.user_arg = data_arg;
+	  rc = lps33hw_enable_interrupt(sensor,
+					 lps33hw_read_interrupt_handler, sensor);
+	  if (rc) {
+	      return rc;
+	  }
+      }
+      else {
```
Comment (vrahane) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: Indentation here as well.

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 3]
```diff
@@ -95,6 +102,51 @@ static const struct sensor_driver g_lps33hw_sensor_driver = {
...
 
+/*
+ * Sensor read after ONE_SHOT conversion
+ */
+#if MYNEWT_VAL(LPS33HW_ONE_SHOT_MODE)
+static void lps33hw_one_shot_read_cb(struct os_event *ev)
+{
+  int rc;
+  struct lps33hw *lps33hw;
+  static struct sensor *sensor;
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: static here does not have add any value

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 4]
```diff
@@ -95,6 +102,51 @@ static const struct sensor_driver g_lps33hw_sensor_driver = {
...
+	  lps33hw->pdd.user_handler = lps33hw->data_func;
+	  lps33hw->pdd.user_arg = &lps33hw->ctx;
+
+	  rc = lps33hw_enable_interrupt(sensor,
+					lps33hw_read_interrupt_handler, sensor);
+      } else {
+	  /* Read once */
+	  struct sensor_press_data spd;
+	  rc = lps33hw_get_pressure(itf, &spd.spd_press);
+	  if(!rc){
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: missing spaces after if and )

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 5]
```diff
@@ -95,6 +102,51 @@ static const struct sensor_driver g_lps33hw_sensor_driver = {
...
+	      spd.spd_press_is_valid = 1;
+	      rc = lps33hw->data_func(sensor, &lps33hw->ctx, &spd, SENSOR_TYPE_PRESSURE);
+	  }
+      }
+  }
+  if (lps33hw->type & SENSOR_TYPE_TEMPERATURE) {
+      struct sensor_temp_data std;
+      rc = lps33hw_get_temperature(itf, &std.std_temp);
+      if(!rc){
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: missing space after if and )

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 6]
```diff
@@ -95,6 +102,51 @@ static const struct sensor_driver g_lps33hw_sensor_driver = {
     .sd_clear_high_trigger_thresh = lps33hw_sensor_clear_high_thresh
 };
 
+/*
+ * Sensor read after ONE_SHOT conversion
+ */
+#if MYNEWT_VAL(LPS33HW_ONE_SHOT_MODE)
+static void lps33hw_one_shot_read_cb(struct os_event *ev)
+{
+  int rc;
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: indentation inconsistent with mynewt style, should be 4

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 7]
```diff
@@ -1025,59 +1080,124 @@ static int
...
-            rc = lps33hw_get_pressure(itf, &spd.spd_press);
-            if (rc) {
-                return rc;
-            }
-
-            spd.spd_press_is_valid = 1;
-
-            rc = data_func(sensor, data_arg, &spd, SENSOR_TYPE_PRESSURE);
-        }
+  int rc = SYS_EINVAL;
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: indentation inconsistent, should be 4

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 8]
```diff
@@ -1025,59 +1080,124 @@ static int
...
+  struct sensor_itf *itf;
+  itf = SENSOR_GET_ITF(sensor);
+  uint8_t rate;
+  rc = lps33hw_get_value(itf, LPS33HW_CTRL_REG1_ODR, &rate);
+  if (rc) {
+      return rc;
+  }
+  struct lps33hw *lps33hw;
+  lps33hw = (struct lps33hw *)SENSOR_GET_DEVICE(sensor);
+  if(rate == LPS33HW_75HZ){
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: missing spaces

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 9]
```diff
@@ -1025,59 +1080,124 @@ static int
...
+	  if (lps33hw->cfg.int_cfg.data_rdy) {
+	      /* Stream read */
+	      lps33hw->pdd.user_handler = data_func;
+	      lps33hw->pdd.user_arg = data_arg;
+	      rc = lps33hw_enable_interrupt(sensor,
+	            lps33hw_read_interrupt_handler, sensor);
+	      if (rc) {
+		  return rc;
+	      }
+	  }
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: else should be on same line as }
} else {

File: hw/drivers/sensors/lps33hw/src/lps33hw.c [hunk 10]
```diff
@@ -1025,59 +1080,124 @@ static int
...
+		  return rc;
+	      }
+	      std.std_temp_is_valid = 1;
+	      rc = data_func(sensor, data_arg, &std,
+			     SENSOR_TYPE_TEMPERATURE);
+	  }
+	  if (!(type & SENSOR_TYPE_TEMPERATURE) && !(type & SENSOR_TYPE_PRESSURE)) {
+	      return SYS_EINVAL;
+	  }
+      }
```
Comment (kasjer) [inline hw/drivers/sensors/lps33hw/src/lps33hw.c]: this closing } was in different place in before, now it's not possible to read temperature without pressure

File: hw/drivers/sensors/lps33thw/src/lps33thw.c [hunk 11]
```diff
@@ -96,6 +102,53 @@ static const struct sensor_driver g_lps33thw_sensor_driver = {
...
 };
 
+/*
+ * Sensor read after ONE_SHOT conversion
+ */
+#if MYNEWT_VAL(LPS33THW_ONE_SHOT_MODE)
+static void lps33thw_one_shot_read_cb(struct os_event *ev){
+  int rc;
+  struct lps33thw *lps33thw;
+  static struct sensor *sensor;
```
Comment (kasjer) [inline hw/drivers/sensors/lps33thw/src/lps33thw.c]: static not needed

File: hw/drivers/sensors/lps33thw/src/lps33thw.c [hunk 12]
```diff
@@ -96,6 +102,53 @@ static const struct sensor_driver g_lps33thw_sensor_driver = {
     .sd_clear_high_trigger_thresh = lps33thw_sensor_clear_high_thresh
 };
 
+/*
+ * Sensor read after ONE_SHOT conversion
+ */
+#if MYNEWT_VAL(LPS33THW_ONE_SHOT_MODE)
+static void lps33thw_one_shot_read_cb(struct os_event *ev){
```
Comment (kasjer) [inline hw/drivers/sensors/lps33thw/src/lps33thw.c]: { should be in the next line

File: hw/drivers/sensors/lps33thw/src/lps33thw.c [hunk 13]
```diff
@@ -96,6 +102,53 @@ static const struct sensor_driver g_lps33thw_sensor_driver = {
...
+
+  if (lps33thw->type & SENSOR_TYPE_PRESSURE) {
+      if (lps33thw->cfg.int_cfg.data_rdy) {
+	  /* Stream read */
+	  lps33thw->pdd.user_handler = lps33thw->data_func;
+	  lps33thw->pdd.user_arg = &lps33thw->ctx;
+
+	  rc = lps33thw_enable_interrupt(sensor,
+					lps33thw_read_interrupt_handler, sensor);
+      }
```
Comment (kasjer) [inline hw/drivers/sensors/lps33thw/src/lps33thw.c]: } else {

File: hw/drivers/sensors/lps33thw/src/lps33thw.c [hunk 14]
```diff
@@ -96,6 +102,53 @@ static const struct sensor_driver g_lps33thw_sensor_driver = {
...
+	      spd.spd_press_is_valid = 1;
+	      rc = lps33thw->data_func(sensor, &lps33thw->ctx, &spd, SENSOR_TYPE_PRESSURE);
+	  }
+      }
+  }
+  if (lps33thw->type & SENSOR_TYPE_TEMPERATURE) {
+      struct sensor_temp_data std;
+
+      rc = lps33thw_get_temperature(itf, &std.std_temp);
+      if(!rc){
```
Comment (kasjer) [inline hw/drivers/sensors/lps33thw/src/lps33thw.c]: spaces missing

File: hw/drivers/sensors/lps33hw/include/lps33hw/lps33hw.h [hunk 15]
```diff
@@ -95,6 +95,13 @@ struct lps33hw {
 #if MYNEWT_VAL(BUS_DRIVER_PRESENT)
     bool node_is_spi;
 #endif
+#if MYNEWT_VAL(LPS33HW_ONE_SHOT_MODE)
+    sensor_type_t type;
+    sensor_data_func_t data_func;
+    struct sensor_read_ctx ctx;
```
[the remark this answers is not in the document]
Comment (reshtitus) [reply hw/drivers/sensors/lps33hw/include/lps33hw/lps33hw.h]: They are different functions and we need both.

**Comment by vrahane** [comment]
I think, everything looks good now.

**Pull request merged**